diff -up kpipewire-6.0.3/src/pipewiresourceitem.cpp.2~ kpipewire-6.0.3/src/pipewiresourceitem.cpp
--- kpipewire-6.0.3/src/pipewiresourceitem.cpp.2~	2024-03-26 16:19:14.000000000 +0100
+++ kpipewire-6.0.3/src/pipewiresourceitem.cpp	2024-03-26 21:32:25.496544004 +0100
@@ -38,8 +38,8 @@ public:
     uint m_nodeId = 0;
     std::optional<uint> m_fd;
     std::function<QSGTexture *()> m_createNextTexture;
-    QScopedPointer<PipeWireSourceStream> m_stream;
-    QScopedPointer<QOpenGLTexture> m_texture;
+    std::unique_ptr<PipeWireSourceStream> m_stream;
+    std::unique_ptr<QOpenGLTexture> m_texture;
 
     EGLImage m_image = nullptr;
     bool m_needsRecreateTexture = false;
@@ -117,7 +117,7 @@ void PipeWireSourceItem::itemChange(QQui
 void PipeWireSourceItem::releaseResources()
 {
     if (window() && (d->m_image || d->m_texture)) {
-        window()->scheduleRenderJob(new DiscardEglPixmapRunnable(d->m_image, d->m_texture.take()), QQuickWindow::NoStage);
+        window()->scheduleRenderJob(new DiscardEglPixmapRunnable(d->m_image, d->m_texture.release()), QQuickWindow::NoStage);
         d->m_image = EGL_NO_IMAGE_KHR;
     }
 }
@@ -182,8 +182,8 @@ void PipeWireSourceItem::refresh()
         }
         d->m_stream->setActive(isVisible());
 
-        connect(d->m_stream.data(), &PipeWireSourceStream::frameReceived, this, &PipeWireSourceItem::processFrame);
-        connect(d->m_stream.data(), &PipeWireSourceStream::stateChanged, this, &PipeWireSourceItem::stateChanged);
+        connect(d->m_stream.get(), &PipeWireSourceStream::frameReceived, this, &PipeWireSourceItem::processFrame);
+        connect(d->m_stream.get(), &PipeWireSourceStream::stateChanged, this, &PipeWireSourceItem::stateChanged);
     }
     Q_EMIT stateChanged();
 }
