diff -up kpipewire-6.0.1/CMakeLists.txt.omv~ kpipewire-6.0.1/CMakeLists.txt
--- kpipewire-6.0.1/CMakeLists.txt.omv~	2024-03-06 01:44:21.250928928 +0100
+++ kpipewire-6.0.1/CMakeLists.txt	2024-03-06 01:44:45.923157698 +0100
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.16)
-project(KPipewire VERSION "6.0.1)
+project(KPipewire VERSION "6.0.1")
 
 set(KF6_MIN_VERSION "6.0.0")
 find_package(ECM ${KF6_MIN_VERSION} REQUIRED NO_MODULE)
