diff -up kpipewire-master/CMakeLists.txt.omv~ kpipewire-master/CMakeLists.txt
--- kpipewire-master/CMakeLists.txt.omv~	2023-05-20 22:16:12.680113365 +0200
+++ kpipewire-master/CMakeLists.txt	2023-05-20 22:16:29.705258558 +0200
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.16)
 project(KPipewire)
 set(PROJECT_VERSION "5.27.80")
-set(PROJECT_VERSION_MAJOR 5)
+set(PROJECT_VERSION_MAJOR 6)
 
 set(KF6_MIN_VERSION "5.240.0")
 find_package(ECM ${KF6_MIN_VERSION} REQUIRED NO_MODULE)
diff -up kpipewire-master/src/CMakeLists.txt.omv~ kpipewire-master/src/CMakeLists.txt
--- kpipewire-master/src/CMakeLists.txt.omv~	2023-05-20 22:15:51.433932260 +0200
+++ kpipewire-master/src/CMakeLists.txt	2023-05-20 22:16:53.532461857 +0200
@@ -162,7 +162,7 @@ install(TARGETS KPipeWire KPipeWireRecor
 ecm_setup_version(PROJECT VARIABLE_PREFIX KPIPEWIRE
                         VERSION_HEADER "${CMAKE_CURRENT_BINARY_DIR}/kpipewire_version.h"
                         PACKAGE_VERSION_FILE "${CMAKE_CURRENT_BINARY_DIR}/KPipeWireConfigVersion.cmake"
-                        SOVERSION 5)
+			SOVERSION ${PROJECT_VERSION_MAJOR})
 
 set(CMAKECONFIG_INSTALL_DIR "${KDE_INSTALL_CMAKEPACKAGEDIR}/KPipeWire")
 install(EXPORT KPipeWireTargets DESTINATION "${CMAKECONFIG_INSTALL_DIR}" FILE KPipeWireTargets.cmake NAMESPACE K::)
